### DISCUSSION TITLE
Como adicionar dependencias no projeto usando clojure-lsp?

### DISCUSSION THREAD

**Discussion by wandersoncferreira** [clj-br/forum#18] [Perguntas e Respostas] [answered]
Eu migrei para o `lsp-mode` recentemente no Emacs e não encontrei uma forma de adicionar dependências com hot-reload via lsp. Isso é possivel de ser feito?

Tanto para projetos lein quanto deps.

**Comment by ericdallo** [comment, marked as answer]
Opa, @wandersoncferreira isso não faz parte da spec oficial do LSP mas nada que não possa ter um custom method do lado do clojure-lsp para isso assim como já existe para algumas outras features, sinta-se livre para abrir uma issue lá que posso no futuro tentar implementar :D

Obs: Lembrando que o LSP usa somente analise estatica e nada relacionado com REPL, então hot-reload ou coisas do genero não funcionam com clojure-lsp

**Reply by wandersoncferreira** [reply]
> Obs: Lembrando que o LSP usa somente analise estatica e nada relacionado com REPL, então hot-reload ou coisas do genero não funcionam com clojure-lsp

Faz sentido mesmo. Eu estou usando o `clj-refactor` com um código custom que eu fiz em elisp para usar o `pomegranate` e incluir a nova dependencia tanto no arquivo deps quanto na seção repl aberta. Funciona bem, mas estava pensando em remover o clj-refactor das dependencias.

**Reply by ericdallo** [reply]
Eu não uso o clj-refactor faz um tempo, pois as features do clojure-lsp ja me satisfazem, mas realmente essa feature eu não tenho costume de usar mas me parece uma boa ideia.
Se pudesse abrir uma issue no clojure-lsp de como espera que funcionasse seria de grande ajuda :)

**Reply by wandersoncferreira** [reply]
> Se pudesse abrir uma issue no clojure-lsp de como espera que funcionasse seria de grande ajuda :)

hoje a noite eu abro uma issue por lá. Valeu!

**Reply by wandersoncferreira** [reply]
@ericdallo https://github.com/clojure-lsp/clojure-lsp/issues/461 :)
